internal/pgd/module.go

```diff
@@ -241,53 +245,50 @@ func (m *Module) applyKeyFuncs(f *jen.File, in pgs.File) error {
 			continue
 		}
 
-		keys := []namedKey{
-			{
-				key:  mext.Partition,
-				name: "PartitionKey",
-			},
-			{
-				key:  mext.Sort,
-				name: "SortKey",
-			},
-			{
-				key:  mext.Version,
-				name: "Version",
-			},
-		}
+		keys := []namedKey{ }
+		// pk, sk, gsi1pk, gsi1sk
+		for _, ck := range mext.Key {
```

Comment:
can this check for repeats?  eg re-using a GSI or PK?  or will that cause a code generatin failure elsewhere?

Reply:
Yes we should explicitly check it. Good call.

Reply:
@pquerna what do you think about getting rid id of the `global_secondary_index` and just increment. So you add multiple keys and the first is the pk, the second is the gsi1, etc.

Reply:
Removed in 3c5ac2d81b2fe6e30564e8499b3f7af1719502b0